modules/renter/hostdb/persist.go

```diff
@@ -0,0 +1,72 @@
+package hostdb
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/NebulousLabs/Sia/persist"
+)
+
+const persistFilename = "hostdb.json"
+
+var saveMetadata = persist.Metadata{
+	Header:  "HostDB Persistence",
+	Version: "0.5",
+}
+
+func (hdb *HostDB) save() error {
+	var data struct {
+		Contracts []hostContract
+	}
+	for _, hc := range hdb.contracts {
+		// to avoid race conditions involving the contract's mutex, copy it
+		// manually into a new object
+		data.Contracts = append(data.Contracts, hostContract{
+			ID:              hc.ID,
+			FileContract:    hc.FileContract,
+			LastRevisionTxn: hc.LastRevisionTxn,
+		})
+	}
+	return persist.SaveFile(saveMetadata, data, filepath.Join(hdb.persistDir, persistFilename))
```

Comment:
the entropy? I'm not sure if it's important, and I'm not sure why it needs to be deterministic. But I'm guessing we want to save it if we're going through the trouble of having it be deterministic in the first place.